src/Components/Facility/SetInventoryForm.tsx

```diff
@@ -50,65 +43,51 @@ export const SetInventoryForm = (props: any) => {
   const { goBack } = useAppHistory();
   const [state, dispatch] = useReducer(inventoryFormReducer, initialState);
   const { facilityId } = props;
-  const dispatchAction: any = useDispatch();
-  const [isLoading, setIsLoading] = useState(false);
   const [data, setData] = useState<Array<InventoryItemsModel>>([]);
   const [currentUnit, setCurrentUnit] = useState<any>();
-  const [facilityName, setFacilityName] = useState("");
 
   const limit = 14;
   const offset = 0;
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-
+  useQuery(routes.getMinQuantity, {
+    pathParams: {
+      facilityId,
+    },
+    prefetch: !!facilityId,
+    onResponse: async ({ data }) => {
       const existingItemIDs: number[] = [];
-      const resMinQuantity = await dispatchAction(
-        getMinQuantity(facilityId, {})
-      );
-
-      resMinQuantity.data.results.map((item: any) =>
-        existingItemIDs.push(item.item_object.id)
-      );
 
-      const res = await dispatchAction(getItems({ limit, offset }));
-
-      if (!status.aborted) {
-        if (res && res.data) {
-          const filteredData = res.data.results.filter(
-            (item: any) => !existingItemIDs.includes(item.id)
-          );
-          setData(filteredData);
-          dispatch({
-            type: "set_form",
-            form: { ...state.form, id: filteredData[0]?.id },
-          });
-        }
-        setIsLoading(false);
+      if (data?.results) {
+        data.results.map((item: any) =>
+          existingItemIDs.push(item.item_object.id)
+        );
       }
-    },
-    [dispatchAction]
-  );
-  useAbortableEffect(
-    (status: statusType) => {
-      fetchData(status);
-    },
-    [fetchData]
-  );
 
-  useEffect(() => {
-    async function fetchFacilityName() {
-      if (facilityId) {
-        const res = await dispatchAction(getAnyFacility(facilityId));
+      await request(routes.getItems, {
+        query: {
+          limit,
+          offset,
+        },
+        onResponse: ({ res, data }) => {
+          if (res && data) {
+            const filteredData = data.results.filter(
+              (item: any) => !existingItemIDs.includes(item.id)
```

Comment:
Avoid using `any` since useQuery/request is type safe.

```suggestion
              (item) => !existingItemIDs.includes(item.id)
```